### PR TITLE
Enable SSSD authentication by default in containers: uyuni changes

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

- Enable SSSD, PAM and LDAP authentication by default
- Persists /etc/sssd folder: all authentication mechanism should be configured using sssd

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: **add explanation. This can't be used if there is a GUI diff**
- No documentation needed: only internal and user invisible changes
- Documentation issue was created: [Link for SUSE Manager contributors](https://github.com/SUSE/spacewalk/issues/new?template=ISSUE_TEMPLATE_DOCUMENTATION.md&labels=documentation&projects=SUSE/spacewalk/31), [Link for community contributors](https://github.com/uyuni-project/uyuni-docs/issues/new).
- API documentation added: please review the Wiki page [Writing Documentation for the API](https://github.com/uyuni-project/uyuni/wiki/Writing-documentation-for-the-API) if you have any changes to API documentation.
- (OPTIONAL) [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pulls)

- [ ] **DONE**

## Test coverage
- No tests
- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/22545

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

File: containers/server-image/Dockerfile [hunk 1]
```diff
@@ -73,16 +78,23 @@ COPY java_agent.yaml /etc/prometheus-jmx_exporter/tomcat/java_agent.yml
 COPY java_agent.yaml /etc/prometheus-jmx_exporter/taskomatic/java_agent.yml
 COPY tomcat_jmx.conf /usr/lib/systemd/system/tomcat.service.d/jmx.conf
 COPY taskomatic_jmx.conf /usr/lib/systemd/system/taskomatic.service.d/jmx.conf
+COPY pam_susemanager /etc/pam.d/susemanager
```
Comment (rjmateus) [inline containers/server-image/Dockerfile:81]: file needs to be readable by tomcat cat. Which access will this file have?

**Review by rjmateus — COMMENTED** [review]
just a few questions, but it looks good to me.

File: containers/server-image/Dockerfile (hunk 1 above)
Comment (mbussolotto) [reply containers/server-image/Dockerfile:81]:
```
-rw-r--r-- 1 root root 164 Apr  9 09:33 /etc/pam.d/susemanager
```
do you think it's fine or is it better to have other permission?

File: containers/server-image/Dockerfile [hunk 2]
```diff
@@ -73,16 +78,23 @@ COPY java_agent.yaml /etc/prometheus-jmx_exporter/tomcat/java_agent.yml
...
 
 RUN chmod -R 755 /usr/bin/timezone_alignment.sh
 
 # Set a warning message for interactive shells
 COPY motd /etc/motd
 RUN echo "cat /etc/motd" >/etc/sh.shrc.local
 
+RUN pam-config -a --sss
```
[the remark this answers is not in the document]
Comment (rjmateus) [reply containers/server-image/Dockerfile:89]: ok, makes sense and looks good to me

File: containers/server-image/Dockerfile (hunk 1 above)
Comment (rjmateus) [reply containers/server-image/Dockerfile:81]: all good. tomcat user only needs to be able to read it, and with that permissions is fine.

**Review by rjmateus — APPROVED** [review]

File: containers/server-helm/server-helm.changes.mbussolotto.pam [hunk 3]
```diff
@@ -0,0 +1 @@
+- Use SSSD authentication in k8s
```
Comment (cbosdo) [inline containers/server-helm/server-helm.changes.mbussolotto.pam:1]:
```suggestion
- Prepare for LDAP authentication using SSSD
```

File: containers/server-image/Dockerfile [hunk 4]
```diff
@@ -73,16 +78,23 @@ COPY java_agent.yaml /etc/prometheus-jmx_exporter/tomcat/java_agent.yml
...
+RUN pam-config -a --ldap
+RUN authselect select sssd --force
+RUN authselect apply-changes
```
Comment (cbosdo) [inline containers/server-image/Dockerfile:92]: Shouldn't those `RUN`s be chained into one? IIRC this reduces the number of image layers.

```suggestion
RUN pam-config -a --sss && \
         pam-config -a --ldap && \
         authselect select sssd --force && \
         authselect apply-changes
```

**Review by cbosdo — APPROVED** [review]
cosmetic suggestions, but otherwise looks good to me

**Pull request merged**